pymc/gp/hsgp_approx.py

```diff
@@ -85,6 +88,67 @@ def calc_basis_periodic(
     return phi_cos, phi_sin
 
 
+def approx_hsgp_hyperparams(x_range: list[float], lengthscale_range: list[float], cov_func: str):
+    """Utility function that uses heuristics to recommend minimum `m` and `c` values,
+    based on recommendations from Ruitort-Mayol et. al.
+
+    In practice, you need to choose `c` large enough to handle the largest lengthscales,
+    and `m` large enough to accommodate the smallest lengthscales.
+
+    NB: These recommendations are based on a one-dimensional GP.
+
+    Parameters
+    ----------
+    x_range : List[float]
+        The range of the input variable on which the GP is going to be evaluated.
+        Should be a list with two elements [x_min, x_max].
+    lengthscale_range : List[float]
+        The range of the lengthscales. Should be a list with two elements [lengthscale_min, lengthscale_max].
+    cov_func : str
+        The covariance function to use. Supported options are "expquad", "matern52", and "matern32".
+
+    Returns
+    -------
+    Tuple[int, float, float]
+        A tuple containing the recommended values for `m`, `c`, and `S`.
+        - `m` : int
+            Number of basis vectors. Increasing it helps approximate smaller lengthscales, but increases computational cost.
+        - `c` : float
+            Scaling factor such that L = c * S, where L is the boundary of the approximation.
+            Increasing it helps approximate larger lengthscales, but may require increasing m.
+        - `S` : float
+            The value of `S`, which is half the range of `x`.
+
+    Raises
+    ------
+    ValueError
+        If either `x_range` or `lengthscale_range` is not in the correct order.
+
+    References
+    ----------
+    - Ruitort-Mayol, G., Anderson, M., Solin, A., Vehtari, A. (2022).
+    Practical Hilbert Space Approximate Bayesian Gaussian Processes for Probabilistic Programming
+    """
+    if (x_range[0] >= x_range[1]) or (lengthscale_range[0] >= lengthscale_range[1]):
+        raise ValueError("One of the boundaries out of order")
+
+    S = (x_range[1] - x_range[0]) / 2
+
+    if cov_func.lower() == "expquad":
+        a1, a2 = 3.2, 1.75
+
+    elif cov_func.lower() == "matern52":
+        a1, a2 = 4.1, 2.65
+
+    elif cov_func.lower() == "matern32":
+        a1, a2 = 4.5, 3.42
+
+    c = max(a1 * (lengthscale_range[1] / S), 1.2)
+    m = int(a2 * c / (lengthscale_range[0] / S))
+
+    return m, c, S
```

Comment:
Just a suggestion: what about returning a [namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) so that we do not depend on the special order of the parameters? something like

```python
HSGPParam = namedtuple('HSGPParam ', ['m', 'c', 'S'])
```

Reply:
Good idea. Done

Reply:
Personally think this is over-engineering, because introducing a new `HSGPParam` type serves no purpose other than to compensate for a perceived defect with the Python language.  I think it was clearer before.

Reply:
No strong opinion here. I just mess up indices so often that I prefer going safe with names instead 🙈

Reply:
yah but I would just expect people to unpack like, `m, c, S = ...`.  

Which makes me think it might make sense (in this or a later PR) to slightly refactor this so that there's a different function,
```python
S = calc_X_radius(X) # or calc_halfwidth or something
```
that could be used later where `_X_mean` was replaced.  Then this function only needs to return `m` and `c` and not also `S`, since that's handled now.